packages/n3h-mock/lib/index.js

```diff
@@ -0,0 +1,289 @@
+    this._ipc.boundEndpoint = (await this._ipc.bind(this._ipcUri))[0]
+
+    log.t('bound to', this._ipc.boundEndpoint)
+  }
+
+  // Received 'message' from IPC: process it
+  _handleIpcMessage (opt) {
+
+    if (opt.name === 'ping') {
+      return
+    }
+
+    log.t('Received IPC message: ', opt)
+
+    let ref
+    let toZmqId
+    if (opt.name === 'json' && typeof opt.data.method === 'string') {
+      switch (opt.data.method) {
+        case 'requestState':
+          this._ipc.send('json', {
+            method: 'state',
+            state: 'ready',
+            id: '42' /*this._p2p.getId()*/,
+            bindings: [] /*this._p2p.getBindings()*/
+          })
+          return
+        case 'connect':
+          // maybe log an error?
+          this._ipc.send('json', {
+            method: 'peerConnected',
+            id: opt.data.address
+          })
+
+          return
+        case 'trackApp':
+          this._track(opt.data.dnaHash, opt.data.agentId, opt.from)
+          return
+        case 'send':
+          ref = this._getMemRef(opt.data.dnaHash)
+
+          if (!(opt.data.toAgentId in this._senders)) {
+            this._ipc.send('json', {
+              method: 'failureResult',
+              dnaHash: opt.data.dnaHash,
+              toAgentId: opt.data.fromAgentId,
+              errorInfo: 'No routing for agent id "' + opt.data.toAgentId + '" aborting send'
+            })
+            return
+          }
+
+          toZmqId = this._senders[opt.data.toAgentId]
+          this._ipc.sendOne(toZmqId, 'json', {
+            method: 'handleSend',
+            _id: opt.data._id,
+            dnaHash: opt.data.dnaHash,
+            toAgentId: opt.data.toAgentId,
+            fromAgentId: opt.data.fromAgentId,
+            data: opt.data.data
+          })
+          return
+        case 'handleSendResult':
+          ref = this._getMemRef(opt.data.dnaHash)
+
+          if (!(opt.data.toAgentId in this._senders)) {
+            log.t('send failed: unknown target node: ' + opt.data.toAgentId)
+            return
+          }
+          toZmqId = this._senders[opt.data.toAgentId]
+
+          if (!(opt.data.toAgentId in this._senders)) {
+            this._ipc.send('json', {
+              method: 'failureResult',
+              dnaHash: opt.data.dnaHash,
+              toAgentId: opt.data.fromAgentId,
+              errorInfo: 'No routing for agent id "' + opt.data.toAgentId + '" aborting handleSendResult'
+            })
+            return
+          }
+          toZmqId = this._senders[opt.data.toAgentId]
+          this._ipc.sendOne(toZmqId, 'json', {
+            method: 'sendResult',
+            _id: opt.data._id,
+            dnaHash: opt.data.dnaHash,
+            toAgentId: opt.data.toAgentId,
+            fromAgentId: opt.data.fromAgentId,
+            data: opt.data.data
+          })
+          return
+        case 'publishDht':
+          this._getMemRef(opt.data.dnaHash).mem.insert({
```

Comment:
heh, this works : ) Strictly speaking, we don't actually need to store the data on the nodejs side, we could just make all the store requests to connected nodes inline here, but that could be an optimization for later.